src/app/frontend/discovery/controller.js

```diff
@@ -25,10 +25,15 @@ export class DiscoveryController {
   constructor(discovery, kdServiceListResource, kdIngressListResource) {
     /** @export {!backendApi.Discovery} */
     this.discovery = discovery;
+
     /** @export {!angular.Resource} */
     this.kdServiceListResource = kdServiceListResource;
+
     /** @export {!angular.Resource} */
     this.kdIngressListResource = kdIngressListResource;
+
+    /** @export {!Array<backendApi.Error>} */
+    this.errors = this.discovery.serviceList.errors.concat(this.discovery.ingressList.errors);
```

Comment:
Maybe this.discovery could have `.errors` or `.warnings` property and it would be merged on the backend? This'd move the logic to the backend and simplify matters here a lot.